clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -728,6 +728,15 @@ public void onFailure(RuntimeException e) {
                     log.trace("Update fetching position to {} for partition {}", nextPosition, completedFetch.partition);
                     subscriptions.position(completedFetch.partition, nextPosition);
                     positionAdvanced = true;
+                    if (partRecords.isEmpty()) {
+                        log.debug(
+                                "Advanced position for partition {} without receiving any user-visible records. " 
+                                        + "This is likely due to skipping over control records in the current fetch, " 
```

Comment:
How about this. First, we can augment the message above to something like this:
```scala
  log.trace("Updating fetch position from {} to {} for partition {} and returning {} records from `poll()`", 
    position, nextPosition, completedFetch.partition, partRecords.size());
```
This gives us enough information in the logs to see which partitions caused the `poll()` to return and it tell us exactly where in the log the aborted transaction/control records exist. Second, maybe we can add back your previous log line, but make it a little more terse and put it at trace level:
```scala
                        log.trace(
                                "Returning empty records from `poll()` since the consumer's position has advanced "
                                        + "for at least one topic partition")
```

Reply:
Sounds good to me 👍